src/plugins/data/public/ui/dataset_selector/dataset_selector.tsx

```diff
@@ -57,15 +58,26 @@ export const DatasetSelector = ({
 
     if (!isMounted.current) return;
 
+    const recentDatasets = datasetService.getRecentDatasets();
     const fetchedDatasets =
       fetchedIndexPatternDataStructures.children?.map((pattern) =>
         typeConfig.toDataset([pattern])
       ) ?? [];
-    setDatasets(fetchedDatasets);
+    const filteredFetchedDatasets = fetchedDatasets.filter(
+      (dataset) => !recentDatasets.some((ds) => ds.id === dataset.id)
+    );
```

Comment:
```suggestion
    const recentDatasetIds = new Set(recentDatasets.map(ds => ds.id));

    const filteredFetchedDatasets = fetchedDatasets.filter(
      dataset => !recentDatasetIds.has(dataset.id)
   );
```
nit: slightly easier to read